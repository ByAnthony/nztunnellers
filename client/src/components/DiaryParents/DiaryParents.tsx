import {
  Parent, Parents,
} from '../../types/tunneller';
import STYLES from '../ProfileDiary/ProfileDiary.module.scss';

  type props = {
    parents: Parents;
  }

export function DiaryParents({ parents }: props) {
  const displayParents = (family: Parents) => {
    if (family.mother !== null && family.father !== null) {
      return (
        <>
          <div className={STYLES['fullwidth-main-card']}>
            Parents
          </div>
          <div className={STYLES['halfwidth-cards-container']}>
            <div className={STYLES['halfwidth-secondary-card']}>
              <div className={STYLES['halfwidth-secondary-card-title']}><p>Mother</p></div>
              <div><span>{ family.mother.name }</span></div>
            </div>
            <div className={STYLES['halfwidth-secondary-card']}>
              <div className={STYLES['halfwidth-secondary-card-title']}><p>Father</p></div>
              <div><span>{ family.father.name }</span></div>
            </div>
          </div>
        </>
      );
    }

    const isMotherOrFather = parents.mother ? 'Mother' : 'Father';

    const displayParent = (parent: Parent) => (
      <div className={STYLES['halfwidth-cards-container']}>
        <div className={STYLES['halfwidth-main-card']}>
          <span>Parent</span>
        </div>
        <div className={STYLES['halfwidth-secondary-card']}>
          <div className={STYLES['halfwidth-secondary-card-title']}><p>{ isMotherOrFather }</p></div>
          <div><span>{ parent.name }</span></div>
        </div>
      </div>
    );

    if (parents.mother !== null && parents.father === null) {
      return displayParent(parents.mother);
    }
    if (parents.mother === null && parents.father !== null) {
      return displayParent(parents.father);
    }
    return null;
  };

  return (
    <>
      { displayParents(parents) }
    </>
  );
}
