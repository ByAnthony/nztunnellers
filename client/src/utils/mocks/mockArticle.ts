import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
