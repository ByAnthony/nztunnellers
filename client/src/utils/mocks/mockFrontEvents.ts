import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
