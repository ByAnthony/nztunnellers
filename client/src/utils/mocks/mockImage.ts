import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'Courtesy of John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook: ImageBook = {
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: 'p. 89',
};

export const mockImageSource: ImageSource = {
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
};

export const mockImage: Image = {
  url: '1/1234.jpg',
  source: mockImageSource,
};
