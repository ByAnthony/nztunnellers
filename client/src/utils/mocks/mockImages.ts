import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook = (overrides: Partial<ImageBook>
    | undefined = undefined): ImageBook => ({
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: '89',
  ...overrides,
});

const mockSource = (overrides: Partial<ImageSource> | undefined = undefined): ImageSource => ({
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
  ...overrides,
});

export const mockImages = (overrides: Partial<Image> | undefined = undefined): Image => ({
  url: '',
  source: mockSource(),
  ...overrides,
});
