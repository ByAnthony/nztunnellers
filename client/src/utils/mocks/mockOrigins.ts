import { Birth, Origins, Parents } from '../../types/tunneller';

export const mockBirth: Birth = {
  date: {
    year: '1886',
    dayMonth: '18 December',
  },
  country: 'New Zealand',
};

export const mockParents: Parents = {
  mother: {
    name: 'Jane Doe',
    origin: 'New Zealand',
  },
  father: {
    name: 'John Doe',
    origin: 'Scotland',
  },
};

export const mockOrigins: Origins = {
  birth: mockBirth,
  parents: mockParents,
  inNzLength: null,
};
