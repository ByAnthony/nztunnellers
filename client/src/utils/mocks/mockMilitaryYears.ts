import {
  Date,
  DeathPlace,
  Death,
  DeathCause,
  Cemetery,
  Medal,
  MilitaryYears,
  Enlistment,
  EmbarkationUnit,
  Training,
  Transport,
  EndOfService,
  Demobilization,
} from '../../types/tunneller';
import { mockFrontEvents } from './mockFrontEvents';

export const mockBritishWarMedal: Medal = {
  name: 'British War Medal',
  country: 'United Kingdom',
  image: 'british-war-medal.png',
  citation: null,
};

export const mockVictoryMedal: Medal = {
  name: 'Victory Medal',
  country: 'United Kingdom',
  image: 'victory-medal.png',
  citation: null,
};

export const mockMedals: Medal[] = [
  mockBritishWarMedal,
  mockVictoryMedal,
];

export const mockDate: Date = {
  year: '1926',
  dayMonth: '13 October',
};

const mockPlace: DeathPlace = {
  location: 'Telegraph Hill',
  town: 'Arras',
  country: 'France',
};

const mockCause: DeathCause = {
  type: 'Killed in action',
  circumstances: 'Dead wounded by shell fire',
};

const mockCemetery: Cemetery = {
  name: 'Faubourg d\'Amiens',
  location: 'Arras',
  country: 'France',
  graveReference: 'IE 18',
};

export const mockDeath: Death = {
  date: mockDate,
  place: mockPlace,
  cause: mockCause,
  cemetery: mockCemetery,
  ageAtDeath: 89,
};

export const mockEnlistment: Enlistment = {
  serial: '1/1000',
  rank: 'Sapper',
  date: mockDate,
  district: 'Canterbury',
  alias: null,
  transferredToTunnellers: null,
};

const mockTraining: Training = {
  date: mockDate,
  location: 'Auckland',
  locationType: 'Military Camp',
};

export const mockEmbarkationUnit: EmbarkationUnit = {
  detachment: 'Main Body',
  training: mockTraining,
  section: 'Section No.2',
  attachedCorps: null,
};

export const mockTransport: Transport = {
  reference: 'HMNZT 56',
  vessel: 'Beata',
  departureDate: mockDate,
  departurePort: 'Auckland',
  arrivalDate: mockDate,
  arrivalPort: 'Plymouth',
};

const mockDemobilization: Demobilization = {
  date: mockDate,
  country: 'New Zealand',
};

const mockEndOfService: EndOfService = {
  deserter: false,
  transferred: null,
  deathWar: null,
  transportNz: mockTransport,
  demobilization: mockDemobilization,
};

export const mockMilitaryYears: MilitaryYears = {
  enlistment: mockEnlistment,
  embarkationUnit: mockEmbarkationUnit,
  transportUk: mockTransport,
  frontEvents: mockFrontEvents,
  endOfService: mockEndOfService,
  medals: mockMedals,
};
