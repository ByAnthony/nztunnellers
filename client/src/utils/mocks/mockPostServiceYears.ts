import { PostServiceYears, PostWarDeath } from '../../types/tunneller';
import { mockDeath } from './mockMilitaryYears';

export const mockAfterWarDeath: PostWarDeath = {
  deathWarInjury: false,
  ...mockDeath,
};

export const mockPostServiceYears: PostServiceYears = {
  death: mockAfterWarDeath,
};
