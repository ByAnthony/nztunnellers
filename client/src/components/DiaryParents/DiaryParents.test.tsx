import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { DiaryParents } from './DiaryParents';
import { mockParents } from '../../utils/mocks/mockOrigins';

const component = (
  <DiaryParents parents={mockParents} />
);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('renders parents when known', () => {
  render(component);

  expect(screen.getByText('Parents')).toBeInTheDocument();
  expect(screen.getByText('Mother')).toBeInTheDocument();
  expect(screen.getByText('Jane Doe')).toBeInTheDocument();
  expect(screen.getByText('Father')).toBeInTheDocument();
  expect(screen.getByText('John Doe')).toBeInTheDocument();
});

test('renders only mother when father unknown', () => {
  const mockComponent = (<DiaryParents parents={{ ...mockParents, father: null }} />);

  render(mockComponent);

  expect(screen.queryByText('Parents')).not.toBeInTheDocument();
  expect(screen.getByText('Parent')).toBeInTheDocument();
  expect(screen.getByText('Mother')).toBeInTheDocument();
  expect(screen.getByText('Jane Doe')).toBeInTheDocument();
  expect(screen.queryByText('Father')).not.toBeInTheDocument();
  expect(screen.queryByText('John Doe')).not.toBeInTheDocument();
});

test('renders only father when father unknown', () => {
  const mockComponent = (<DiaryParents parents={{ ...mockParents, mother: null }} />);

  render(mockComponent);

  expect(screen.queryByText('Parents')).not.toBeInTheDocument();
  expect(screen.getByText('Parent')).toBeInTheDocument();
  expect(screen.queryByText('Mother')).not.toBeInTheDocument();
  expect(screen.queryByText('Jane Doe')).not.toBeInTheDocument();
  expect(screen.getByText('Father')).toBeInTheDocument();
  expect(screen.getByText('John Doe')).toBeInTheDocument();
});

test('does not render the component when parents are unknown', () => {
  const { container } = render(
    <DiaryParents parents={{
      ...mockParents,
      mother: null,
      father: null,
    }}
    />,
  );

  expect(container).toBeEmptyDOMElement();
});
