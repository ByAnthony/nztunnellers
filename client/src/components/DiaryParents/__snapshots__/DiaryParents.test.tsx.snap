// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="fullwidth-main-card"
  >
    Parents
  </div>
  <div
    class="halfwidth-cards-container"
  >
    <div
      class="halfwidth-secondary-card"
    >
      <div
        class="halfwidth-secondary-card-title"
      >
        <p>
          Mother
        </p>
      </div>
      <div>
        <span>
          Jane Doe
        </span>
      </div>
    </div>
    <div
      class="halfwidth-secondary-card"
    >
      <div
        class="halfwidth-secondary-card-title"
      >
        <p>
          Father
        </p>
      </div>
      <div>
        <span>
          John Doe
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
